### PR TITLE
Documentation: Add initial document about the Kernel graphics subsystem

### PR DESCRIPTION
**Pull request by supercomputer7**
This is an attempt of me trying to document the recent development of what came to be the kernel graphics subsystem.
This is only the start, and future PRs might address things like explaining specific drivers, or reasons to usage of spinlocks vs mutexes in various places.

Fixes, notes and suggestions are always appreciated :)

### CODE REVIEW

File: Documentation/Kernel/GraphicsSubsystem.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+## Responsibilities
+
+* Provide a convenient interface to all supported video hardware in the Kernel.
+* Manage 3D rendering on supported hardware.
+
+## Current Limitations and Future features?
+
+* No locking on who can do `mmap` on DisplayConnector devices currently, which can
+lead to malicious applications to "fight" with WindowServer on what is shown to the user
```
Comment (AtkinsSJ) [inline Documentation/Kernel/GraphicsSubsystem.md:16]:
```suggestion
lead to malicious applications "fighting" with WindowServer on what is shown to the user
```

File: Documentation/Kernel/GraphicsSubsystem.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+the fact that we essentially can take VRAM access from the WindowServer at anytime we want,
+without the WindowServer being aware of this so it can still function in the background.
+2. We need to know the maximum dimensions of the framebuffers when initializing the device
+and creating the DisplayConnector device at runtime. This happens because we map all the possible
+pages of VRAM framebuffer at that time, and also reserve the same amount of pages in usable
+physical memory space, so we could reserve the contents of VRAM between the switch
+from graphics mode to console mode and vice-versa.
+
+The actual implementation is quite simple, yet powerful enough to let everyone
+to live comfortably - each DisplayConnector device is backed by a special VMObject (VMObject is
```
Comment (AtkinsSJ) [inline Documentation/Kernel/GraphicsSubsystem.md:98]:
```suggestion
live comfortably - each DisplayConnector device is backed by a special VMObject (VMObject is
```

**Comment by AtkinsSJ** [comment]
Let's not let this go stale!

**Comment by supercomputer7** [comment]
I still need to finish this :)

**Comment by supercomputer7** [comment]
Let's try this again :)

**Pull request merged**